src/layouts/components/Editor/Iframe.ts

```diff
@@ -0,0 +1,118 @@
+import { Node, nodeInputRule, nodePasteRule } from "@tiptap/core"
+import { ReactNodeViewRenderer } from "@tiptap/react"
+import _ from "lodash"
+
+import { LegacyBlockView } from "./LegacyBlockView"
+
+export interface IframeOptions {
+  allowFullscreen: boolean
+  HTMLAttributes: {
+    [key: string]: any
+  }
+}
+
+declare module "@tiptap/core" {
+  interface Commands<ReturnType> {
+    iframe: {
+      /**
+       * Add an iframe
+       */
+      setIframe: (options: { src: string }) => ReturnType
+    }
+  }
+}
+
+export default Node.create<IframeOptions>({
+  name: "iframe",
+
+  group: "block",
+  atom: true,
+
+  draggable: true,
+
+  defining: true,
+
+  addOptions() {
+    return {
+      allowFullscreen: true,
+      HTMLAttributes: {
+        class: "iframe-wrapper",
+      },
+    }
+  },
+
+  addAttributes() {
+    return {
+      src: {
+        default: null,
+      },
+      frameborder: {
+        default: 0,
+      },
+      allowfullscreen: {
+        default: this.options.allowFullscreen,
+        parseHTML: () => this.options.allowFullscreen,
+      },
+      width: {
+        default: null,
+      },
+      height: {
+        default: null,
+      },
+      style: {
+        default: null,
+      },
+    }
+  },
+
+  parseHTML() {
+    return [
+      {
+        tag: "iframe",
+      },
+    ]
+  },
+
+  renderHTML({ HTMLAttributes }) {
+    return ["div", this.options.HTMLAttributes, ["iframe", HTMLAttributes]]
+  },
+
+  addCommands() {
+    return {
+      setIframe: (options: { src: string }) => ({ tr, dispatch }) => {
+        const { selection } = tr
+        const node = this.type.create(options)
+
+        if (dispatch) {
+          tr.replaceRangeWith(selection.from, selection.to, node)
+        }
+
+        return true
+      },
+    }
+  },
+
+  addPasteRules() {
+    return [
+      nodePasteRule({
+        find: /<iframe (.*)><\/iframe>/g,
+        type: this.type,
+        getAttributes: (match) => {
+          const srcIndex = match.input?.search("src=")
```

Comment:
hmm am uncomfortable with this regex parsing :( Are we controlling the inputs here ah? eg if user puts `src = blah` this will fail + now we introduce stricter syntax rules when developing tiptap.

Reply:
it'll just be outputted as the most general matching node, which is `Text` here, so the user doesn't lose info.

i'm not inclined to make this change _now_ - most users copy paste their iframe without editing; this is just to allow them a view of their iframe. this seems like an edge case (user actually edits their iframe, specifically the `src` property) which we can deal with when it happen.

do feel free to add this in/a stricter regex if you want it to happen that way. fwiw, on my end, i think it's good to change but maybe not the best use of time now